content/docs/instrumenting/http_sd.md

```diff
@@ -0,0 +1,96 @@
+---
+title: HTTP SD
+sort_rank: 7
+---
+
+# Writing HTTP Service Discovery
+
+Prometheus provides a generic [HTTP Service Discovery](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#http_sd_config),
+that enables it to discover targets over an HTTP endpoint.
+
+The HTTP Service Discovery is complimentary to the supported service
+discovery mechanisms, and is an alternative to [File-based Service Discovery](https://prometheus.io/docs/guides/file-sd/#use-file-based-service-discovery-to-discover-scrape-targets).
+
+## Comparison between File-Based SD and HTTP SD
+
+Here is a table comparing our two generic Service Discovery implementations.
+
+| Item | File SD | HTTP SD |
+| ---- | ------- | ------- |
+| Event Based | Yes, via inotify | No |
+| Update frequency | Instant, thanks to inotify | Following refresh_interval |
+| Format | Yaml or JSON | JSON |
+| Transport | Local file | HTTP/HTTPS |
+| Security | File-Based security | TLS, Basic auth, Authorization header, OAuth2 |
+
+## Requirements of HTTP SD endpoints
```

Comment:
I wonder if we should call this v1 or 1.0.0 and use RFC 2119 language

Reply:
We can version later. At this point, this is not even released. I would refrain from versioning or being too strict here. It is almost rfc2119, we could just turn lowercases into uppercases later on.

Reply:
I propose we do RFC2119 and versioning in the compliance repository.